### ISSUE TITLE
bytes 0.5 compatibility

### ISSUE THREAD

**Issue by Fabian-Gruenbichler** [NLnetLabs/bcder#42]
I'm currently in the process of updating packaged crates for futures 0.3/tokio 0.2 compatibility [in Debian](https://salsa.debian.org/rust-team/debcargo-conf/merge_requests/73), which also entails updating to bytes 0.5.

Unfortunately bytes 0.4->0.5 does entail some breaking changes, and bcder's Captured/CapturedWriter are affected as well:

call in Captured::extend
```
error[E0599]: no method named `extend_from_slice` found for type `&'a mut bytes::bytes::Bytes` in the current scope
   --> src/captured.rs:197:16
    |
197 |         self.0.extend_from_slice(buf);
    |                ^^^^^^^^^^^^^^^^^ help: there is a method with a similar name: `clone_from_slice`
```

I am not sure whether this means that Captured needs to switch over to BytesMut or whether there is a simpler solution. https://github.com/tokio-rs/bytes/issues/368 and https://github.com/tokio-rs/bytes/issues/350 might also be related.

**Comment by partim** [comment]
Just switching to `BytesMut` won’t do, since then there will be lots of copying on cloning – `Captured` currently promises cheap copies. We might have to have a separate `CapturedMut`, but I need to look at the code that uses this stuff to see how awkward that would be.

**Comment by Fabian-Gruenbichler** [comment]
Obviously you know your code base better than me, and I am happy if I don't have to dig in further and can leave it in your hands instead :) it will probably take at least a few weeks until this actual starts to block anything on the Debian side, so no need to hurry (yet ;-) ).

**Comment by partim** [comment]
This is now solved via #43, #46, and #47.
